### PR TITLE
 ci: Add concurrency and remove redundant lines

### PR DESCRIPTION
**Pull request by avinal**
## Proposed changes 

- Add concurrency
- Add workflow dispatch
- Reformat for better readability
- Remove redundant lines
- Enable parallel builds
- Add pull request build
- Enable only master branch for CI

----
Signed-off-by: Avinal Kumar <avinal.xlvii@gmail.com>

### CODE REVIEW

**Comment by pktiuk** [comment]
Can you create new branches on this repo?

**Review by pktiuk — APPROVED** [review]
LGTM

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,67 +1,71 @@
...
 
-on: [push]
+concurrency:
+    group: build-${{ github.head_ref }}
+    cancel-in-progress: true
+
+on:
+    push:
+        branches:
+            - master
```
Comment (pktiuk) [inline .github/workflows/main.yml:10]: Why do we launch it only with pushes to master?

Comment (avinal) [reply .github/workflows/main.yml:10]: With the last setup every workflow was running twice but only when those pull request were opened from this repo only. So the idea is, either you open a PR from a fork or this repo itself. But in either case you will be opening a PR to add changes, so exactly one build will run. And a final build will run when you merge the pull request.

**Comment by avinal** [comment]
> Can you create new branches on this repo?

Yes

File: .github/workflows/main.yml (hunk 1 above)
Comment (avinal) [reply .github/workflows/main.yml:10]: Also, now we wil be able to run it on demand too with workflow dispatch feature. So if nedded you can always trigger build for a different branch.

**Pull request merged**